raiden/smart_contracts/ChannelManagerContract.sol

```diff
@@ -124,7 +128,7 @@ contract ChannelManagerContract {
         uint caller_index;
         uint partner_index;
 
-        (channel_address, has_channel, caller_index, partner_index) = getChannelWith(partner);
+        (channel_address, has_channel, caller_index, partner_index) = getChannelAndIndices(partner);
```

Comment:
Either remove the `getChannelAndIndices` or fix the array iteration in that function. Removing is an option and this could be done:
```
bytes32 party_hash = partyHash(msg.sender, partner);
data.channel_addresses[party_hash] == 0x0
```